wgsl/index.bs

```diff
@@ -1166,15 +1168,46 @@ allows them to naturally use values defined in the loop body.
 </div>
 * <2> The continue construct is placed at the end of the `loop`
 
-### Other Looping Constructs ### {#other-looping}
+### For statement ### {#for-statement}
 
-Note: This is *proposed* but not in the [SHORTNAME] spec yet
+<pre class='def'>
+for_stmt
+  : FOR PAREN_LEFT for_header PAREN_RIGHT BRACKET_LEFT statements BRACKET_RIGHT
+
+for_header
+  : variable_stmt? SEMICOLON logical_or_expression? SEMICOLON assignment_stmt?
+</pre>
 
 The `for(var i : i32 = 0; i < 4; i = i + 1) {}` statement is syntactic sugar on
-top of the [[#loop-statement]].
+top of the [[#loop-statement]]. The `for` transforms into loop as:
+
+<div class='example' heading="For to Loop transformation">
+  <xmp>
+    for(var i : i32 = 0; i < 4; i = i + 1) {
+      if (a == 0) {
+        continue;
+      }
+      a = a + 2;
+    }
 
-The `while(i < 4) {}` is syntactic sugar on top of the [[#loop-statement]]
-where there is no `continuing` construct.
+    Converts too:
+
+    # TODO(dsinclair): Wrap in a block when/if we get lexical scoping
+    var i : i32 = 0;
+    loop {
+      break unless (i < 4);
```

Comment:
oh, this requires some thinking to get used to

Reply:
Just like you learned the original C looping idiom.  It's an idiom you'll work through once and remember later.

Writing it this way it's clear there's a direct correspondence.
Also, if people are writing comparisons with floating point numbers, e.g. a convergence criterion  err< epsilon, then we avoid flipping the sense of the comparison and messing up the NaN behaviour.